### PR TITLE
Fix/git sync depth

### PR DESCRIPTION
**Pull request by mgorsk1**
This PR resolves bug https://github.com/apache/airflow/issues/9093

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/config_templates/config.yml [hunk 1]
```diff
@@ -1945,6 +1945,14 @@
       type: string
       example: ~
       default: ""
+    - name: git_sync_depth
+      description: |
+        Use a shallow clone with a history truncated to the specified number of commits.
+        0 - do not use shallow clone.
+      version_added: ~
```
Comment (potiuk) [inline airflow/config_templates/config.yml]: I like the change - @kaxil should we cherry-pick to 1.10.11 ? If so then this version here should be updated.

**Review by potiuk — COMMENTED** [review]
I like the change but we need to define the version at which it will be added.

File: airflow/config_templates/config.yml (hunk 1 above)
Comment (kaxil) [reply airflow/config_templates/config.yml]: Yeah, we should, looks like a useful change.

Comment (kaxil) [inline airflow/config_templates/config.yml]:
```suggestion
      version_added: 1.10.11
```

**Review by potiuk — APPROVED** [review]

**Comment by mgorsk1** [comment]
thanks @potiuk @kaxil for prompt response !

**Pull request merged**